src/AppInstallerCommonCore/UserSettings.cpp

```diff
@@ -199,6 +200,51 @@ namespace AppInstaller::Settings
         WINGET_VALIDATE_PASS_THROUGH(EFExport)
         WINGET_VALIDATE_PASS_THROUGH(TelemetryDisable)
         WINGET_VALIDATE_PASS_THROUGH(EFRestSource)
+
+        WINGET_VALIDATE_SIGNATURE(InstallScopePreference)
+        {
+            static constexpr std::string_view s_scope_user = "user";
+            static constexpr std::string_view s_scope_machine = "machine";
+
+            if (Utility::CaseInsensitiveEquals(value, s_scope_user))
+            {
+                return ScopePreference::User;
+            }
+            else if (Utility::CaseInsensitiveEquals(value, s_scope_machine))
+            {
+                return ScopePreference::Machine;
+            }
+
+            return {};
```

Comment:
>{} [](start = 19, length = 2)

Do we have a way to output warnings for unrecognize settings?

Reply:
Validate takes care of that when it gets a nullopt back from these functions.

---
In reply to: [602807223](https://github.com/microsoft/winget-cli/pull/819#discussion_r602807223) [](ancestors = 602807223)